src/pages/index.js

```diff
@@ -10,48 +12,131 @@ import Section from "../components/Section.js";
+// API
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      createCard(cardData);
-    },
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "ebfbe580-59e8-4623-9d1e-5edf14608279",
+    "Content-Type": "application/json",
   },
-  "gallery__cards"
-);
+});
+
+// SECTION
 
-section.renderItems();
+let section;
+
+api
+  .getInitialCards()
+  .then((res) => {
+    section = new Section(
+      {
+        items: res,
+        renderer: (cardData) => {
+          createCard(cardData);
+        },
+      },
+      "gallery__cards"
+    );
+    section.renderItems();
+    console.log(res);
+  })
+  .catch(console.error);
+
+api
+  .loadInfo()
+  .then((res) => {
+    console.log(res);
+    userinfo.setUserInfo(res.name, res.about);
+    userinfo.setAvatar(res.avatar);
+  })
+  .catch(console.error);
 
 // POPUP WITH IMAGE
 
 const popupWithImage = new PopupWithImage("#picture-modal");
 
 popupWithImage.setEventListeners();
 
+// SAVE BUTTON TEXT CHANGE FUNCTION FOR FORMS
+
+function setButtonText(buttin, text) {
+  buttin.textContent = text;
+}
+
 // CARD
 
 function handleImageClick(cardData) {
   popupWithImage.open(cardData);
 }
 
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   section.addItem(card.generateCard());
 }
 
 function handleAddCardSubmit(inputValues) {
-  createCard(inputValues);
-  variables.profileAddForm.reset();
-  addFormValidator.toggleButtonState();
+  setButtonText(variables.addModalButton, "Saving...");
+  api.addNewCard(inputValues).then((res) => {
+    console.log(res);
+    // create a new card
+    createCard(res);
+    // render it
+    variables.profileAddForm.reset();
+    // toggle button state
+    addFormValidator.toggleButtonState();
+    // close modal
+    setButtonText(variables.addModalButton, "Save");
+  });
   addPopup.close();
 }
 
+// HANDLE DELETE MODAL
+
+const deleteModal = new PopupWithDelete(variables.deleteCardClass);
+deleteModal.setEventListeners();
+
+function handleDeleteClick(card) {
+  // open the modal
+  deleteModal.open();
+  // set the submit action
+  deleteModal.setSubmitAction(() => {
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        deleteModal.close();
+        card.handleDelete();
+      })
+      .catch(console.error);
+  });
+}
+
+// CARD LIKE
+
+function handleLikeClick(card) {
+  api
+    .updateLike(card._id, card.isLiked())
+    .then(() => {
+      card.handleLike();
+      // card.updateLikeView();
+      console.log(card);
+    })
+    .catch(console.error);
+  // handle changing the button's state in a .then after the successful response
+}
+
 // FORM VALIDATOR
 
 const addFormValidator = new FormValidator(config, variables.profileAddModal);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store the validator using the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```